.bcr/presubmit.yml

```diff
@@ -7,5 +7,7 @@ bcr_test_module:
       working_directory: examples/bzlmod/hello_world
       name: "Run test module"
       platform: ${{ platform }}
+      run_targets:
+        - "//third-party:vendor"
```

Comment:
In CI this is succeeding on Linux (_"bzlmod BCR presubmit on :ubuntu: 20.04 LTS (OpenJDK 11, gcc 9.4.0)"_) but failing on macOS (_"bzlmod BCR presubmit on :darwin: (OpenJDK 11, Xcode)"_).

```console
/private/var/tmp/_bazel_buildkite/4894c9dc50e84673867cbd80d0914a01/external/rules_rust~override~rust~rust_host_tools/bin/cargo: /private/var/tmp/_bazel_buildkite/4894c9dc50e84673867cbd80d0914a01/external/rules_rust~override~rust~rust_host_tools/bin/cargo: cannot execute binary file

ERROR: /Users/buildkite/builds/bk-imacpro-7/bazel/rules-rust-rustlang/examples/bzlmod/hello_world/third-party/BUILD.bazel:3:14: //third-party:vendor depends on @@rules_rust~override~cargo_bazel_bootstrap~cargo_bazel_bootstrap//:binary in repository @@rules_rust~override~cargo_bazel_bootstrap~cargo_bazel_bootstrap which failed to fetch. no such package '@@rules_rust~override~cargo_bazel_bootstrap~cargo_bazel_bootstrap//': Process exited with code '126'
```

Reply:
It looks like it's trying to use a Linux `cargo` binary to build `cargo-bazel` - I repro locally... See https://github.com/bazelbuild/rules_rust/pull/2021#issuecomment-1871381693

cc @matts1 @scentini

Reply:
It looks like the `MODULE.bazel.lock` contains platform-specific information, which gets re-used if nothing else has changed in the repo:

```
          "rust_host_tools": {
            "bzlFile": "@@rules_rust~override//rust:repositories.bzl",
            "ruleClassName": "rust_toolchain_tools_repository",
            "attributes": {
              "name": "rules_rust~override~rust~rust_host_tools",
              "exec_triple": "x86_64-unknown-linux-gnu",
              "target_triple": "x86_64-unknown-linux-gnu",
              "allocator_library": "@rules_rust//ffi/cc/allocator_library",
              "dev_components": false,
              "edition": "2021",
              "rustfmt_version": "nightly/2023-12-07",
              "sha256s": {},
              "urls": [
                "https://static.rust-lang.org/dist/{}.tar.gz"
              ],    
              "version": "1.74.1"
            }     
          },
```

Reply:
Looks like we need to do something like this: https://github.com/bazelbuild/rules_python/pull/1433/files

Reply:
Good find. See also https://github.com/bazelbuild/rules_python/issues/1643. Otherwise, according to https://github.com/bazelbuild/rules_python/pull/1433#issuecomment-1772352358, you and all downstream users need to regenerate their lockfiles separately from each platform they support. For example I, on Linux, would not be able to generate a lockfile that passes checks on macOS.

Reply:
Started a thread on Slack: https://bazelbuild.slack.com/archives/C014RARENH0/p1703787329091939 - hopefully there's a recommendation we can follow :)

Reply:
I tested this in https://github.com/bazelbuild/rules_rust/pull/2374 - it works, but looks like it's going to churn the lockfile every time someone switches platform. Hopefully we can get some useful pointers soon :)

Reply:
Yeah, for now let's just delete + .gitignore the `MODULE.bazel.lock` file in examples/bzlmod/hello_world and be happy

Reply:
I believe there's a way  to ensure that you don't use the lockfile for specific module extensions / repository rules. Don't know how though. @Wyverald should know, he implemented the lockfile.